bin/tag_release.nim

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env -S nim r --verbosity:0 --skipParentCfg:on
+
+## Running this file will:
+## 1. Switch to the `main` branch, if there are no uncommitted changes (exiting
+##    otherwise).
+## 2. Pull the upstream `main` branch from github.com/exercism/configlet.
+## 3. Check that the local `main` is not ahead of upstream.
+## 4. Check that the most recent commit on `main` is an untagged release commit.
+## 5. Prompt the user to tag the commit and push the tag to exercism/configlet,
+##    which will trigger a build job and create a draft release.
+import std/[strformat, strscans, strutils]
+import "."/bump_version
+
+proc checkRepoIsInPreTagState: string =
+  ## Returns the to-be-tagged version.
+  checkRepoState()
+  let commitTitle = execAndCheck(GitLog, ["-n1", "--format='%s'"])
+  let (isMatch, version, _) = commitTitle.scanTuple("release: $+ (#$+i)$.")
+  if isMatch:
+    let existingTag = execAndCheck(GitTag, ["--points-at"])
+    if existingTag.len > 0:
+      error(&"the commit has already been tagged: {existingTag}")
+    result = version
+  else:
+    error(&"the most recent commit on `main` is not a release commit:\n    {commitTitle}")
+
+proc promptToTagAndPush(version: string) =
+  while true:
+    stderr.write "tag the current commit and push the tag to exercism/configlet? ([y]es/[n]o) "
+    case stdin.readLine().toLowerAscii()
+    of "y", "yes":
+      discard execAndCheck(GitTag, ["-a", "-m", version, version])
+      let remote = getGitHubRemoteName("exercism", "configlet")
+      discard execAndCheck(GitPush, [remote, version])
```

Comment:
Currently if the push fails, the commit will still be tagged locally, and re-running the script will fail due to line 22. 

Maybe auto-remove the tag if the push fails?

Reply:
Yep, that sounds good.

Reply:
Attempted to implement: https://github.com/exercism/configlet/pull/531/commits/29f8517da757f7e9173127d29f19d7f47c6ee992

Reply:
LGTM